packages/phpstan-rules/src/Rules/CheckControllerRepositoryLayerRule.php

```diff
@@ -0,0 +1,232 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Symplify\PHPStanRules\Rules;
+
+use Nette\Utils\Strings;
+use PhpParser\Node;
+use PhpParser\Node\Identifier;
+use PhpParser\Node\Name\FullyQualified;
+use PhpParser\Node\Stmt\Class_;
+use PhpParser\Node\Stmt\Property;
+use PhpParser\NodeFinder;
+use PHPStan\Analyser\Scope;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Symplify\PHPStanRules\Tests\Rules\CheckControllerRepositoryLayerRule\CheckControllerRepositoryLayerRuleTest
+ */
+final class CheckControllerRepositoryLayerRule extends AbstractSymplifyRule
+{
+    /**
+     * @var string
+     */
+    public const ERROR_MESSAGE = '%s type not allowed to use %s type as dependency, use %s instead';
+
+    /**
+     * @var string
+     * @see https://regex101.com/r/x1GflV/1
+     */
+    private const ENTITYMANAGER_REGEX = '#EntityManager#i';
+
+    /**
+     * @var string
+     * @see https://regex101.com/r/62rngZ/2
+     */
+    private const NOT_ENTITYMANAGER_REGEX = '#(EntityManager)[^\1]*#i';
+
+    /**
+     * @var string
+     * @see https://regex101.com/r/Azledf/2
+     */
+    private const CONTROLLER_REGEX = '#(Controller$)|\b(Controller)\b#';
+
+    /**
+     * @var string
+     * @see https://regex101.com/r/AQG06A/2
+     */
+    private const REPOSITORY_REGEX = '#(Repository$)|\b(Repository)\b#';
+
+    /**
```

Comment:
The tests and implementation looks good. I'm thinking of how to improve configuration, so we don't have to make specific rules for every case in the future.

If we make this configurable, it could be generic solution to dependency management.


E.g. to forbid `*EntityManger` dependency in `*Controller` class:

```yaml
services:
    -
        class: CheckDependencyMatrixRule
        arguments: 
            frobiddenMatrix:
                '*Controller':
                     - '*EntityManager'
```

Reply:
I will try

Reply:
implemented.